### PR TITLE
core: generate host veth name early in the stack

### PR DESCRIPTION
**Pull request by flouthoc**
get host mac early for usage

### CODE REVIEW

**Comment by baude** [comment]
LGTM

File: src/network/core.rs [hunk 1]
```diff
@@ -110,6 +114,16 @@ impl Core {
...
+        let host_veth_mac = match core_utils::CoreUtils::get_interface_address(&host_veth_name) {
+            Ok(addr) => addr,
+            Err(err) => {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    format!("failed to get host veth mac: {}", err),
+                ))
+            }
+        };
+        debug!("Host veth mac: {:?}", host_veth_mac);
```
Comment (Luap99) [inline src/network/core.rs:126]: Why do we need the host veth mac?

Comment (flouthoc) [reply src/network/core.rs:126]: @Luap99 I think @baude needs it for some of his experiments.

**Pull request merged**